src/features/filter.ts

```diff
@@ -1,12 +1,20 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { FilterBy } from '../types/FilterBy';
 
 const initialState = {
   query: '',
-  status: 'all',
+  status: FilterBy.All,
 };
 
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    filterStatus(state, { payload }: PayloadAction<FilterBy>) {
```

Comment:
```suggestion
    setStatus(state, { payload }: PayloadAction<FilterBy>) {
```
this action sets the filter status, so should be named accordingly
Same below